source/manuals/programming-languages/java.html.md.erb

```diff
@@ -91,6 +93,16 @@ var checkResponse = service.getCheckResponse();
 
 Be mindful that `var` hides the type of the variable. If the variable name makes the type obvious, this usually isn’t a problem. But if it is not clear from either the variable name or the right-hand side of the assignment, it might be better to explicitly write the type.
 
+If you use `var` in place of a generic type, this will usually prevent you from using the so-called diamond operator (itself a form of type inference) on the right-hand side of the assignment:
+
+```java
+// Java 9
+Set<String> usernames = new HashSet<>();
+
+// Java 10+
+var usernames = new HashSet<String>();
```

Comment:
```suggestion
var usernames = new HashSet<String>();
// Note that:
var usernames = new HashSet<>(); // will compile but is equivalent to var usernames = new HashSet<Object>();
```

Reply:
Updated the example to have something similar.